### PR TITLE
Feature/106/integration of candidate list

### PR DESCRIPTION
**Pull request by Dmitruk89**
screen:
Снимок экрана 2021-12-01 в 16 17 29
 1. Create filter method with http-options in api-service.
 2. Create datasource class to bind mat-table with backend data.
 3. Bind mat-table, pagination, sorting, and filtering with backend.

### CODE REVIEW

File: src/app/components/appoint-interview-popup/appoint-interview-popup.component.ts [hunk 1]
```diff
@@ -54,9 +54,9 @@ export class AppointInterviewPopupComponent implements OnInit {
       thirdCtrl: ['', Validators.required],
     });
 
-    setTimeout(() => {
-      this.modalWindowService.modalWindow.next('candidates card');
-    }, 200);
+    // setTimeout(() => {
+    //   this.modalWindowService.modalWindow.next('candidates card');
+    // }, 200);
```
Comment (mhasilau) [inline src/app/components/appoint-interview-popup/appoint-interview-popup.component.ts:59]: I use this

File: src/app/components/candidate-table/candidate-table/candidate-table.component.html [hunk 2]
```diff
@@ -5,39 +5,46 @@
...
         </mat-form-field>
     </div>
     <button class="btn filled-btn">{{ 'candidateList.request' | translate }}</button>
+    </form>
 </mat-expansion-panel>
-<mat-form-field appearance="standard">
+
+<!-- TODO: decide do we need such filter as we have filtering by sending request? -->
+
+<!-- <mat-form-field appearance="standard">
```
Comment (mhasilau) [inline src/app/components/candidate-table/candidate-table/candidate-table.component.html:41]: we till need this?

File: src/app/components/candidate-table/candidate-table/candidate-table.component.ts [hunk 3]
```diff
@@ -42,30 +79,78 @@ export class CandidateTableComponent implements OnInit {
...
+  // applyFilter(event: Event): void {
+  //   const filterValue = (event.target as HTMLInputElement).value;
+  //   console.log(filterValue);
+  //   this.dataSource.filter = filterValue.trim().toLowerCase();
+  //   console.log(this.dataSource);
+  //   if (this.dataSource.paginator) {
+  //     this.dataSource.paginator.firstPage();
+  //   }
+  // }
+
```
Comment (mhasilau) [inline src/app/components/candidate-table/candidate-table/candidate-table.component.ts:148]: and this

File: src/app/components/candidate-table/candidate-table/candidate-table.component.ts [hunk 4]
```diff
@@ -88,6 +173,18 @@ export class CandidateTableComponent implements OnInit {
...
+    this.loadCandidatesPage();
+    this.paginator.firstPage();
+  }
+
+  // TODO: getting selected candidates:
+
+  // selectLocation(event: any): void {
+  //   console.log(event);
+  //   console.log(this.selectLocation);
+  // }
```
Comment (mhasilau) [inline src/app/components/candidate-table/candidate-table/candidate-table.component.ts:189]: and this

**Review by Nicolai-boiko — APPROVED** [review]

**Review by vstrabko — APPROVED** [review]

File: src/app/components/candidate-table/candidate-table/candidate-table.component.html (hunk 2 above)
Comment (Dmitruk89) [reply src/app/components/candidate-table/candidate-table/candidate-table.component.html:41]: resolved

File: src/app/components/candidate-table/candidate-table/candidate-table.component.ts (hunk 3 above)
Comment (Dmitruk89) [reply src/app/components/candidate-table/candidate-table/candidate-table.component.ts:148]: resolved

File: src/app/components/candidate-table/candidate-table/candidate-table.component.ts (hunk 4 above)
Comment (Dmitruk89) [reply src/app/components/candidate-table/candidate-table/candidate-table.component.ts:189]: resolved

**Review by mhasilau — APPROVED** [review]

**Pull request merged**